app/models/manageiq/providers/vmware/infra_manager/vm/operations/guest.rb

```diff
@@ -28,10 +28,11 @@ module ManageIQ::Providers::Vmware::InfraManager::Vm::Operations::Guest
       unsupported_reason_add(:reset, unsupported_reason(:control)) unless supports?(:control)
       unsupported_reason_add(:reset, _("The VM is not powered on")) unless current_state == "on"
     end
-  end
 
-  def validate_standby_guest
-    validate_vm_control_powered_on
+    supports :standby_guest do
+      unsupported_reason_add(:standby_guest, unsupported_reason(:control)) unless supports?(:control)
+      unsupported_reason_add(:standby_guest, _("The VM is not powered on")) unless current_state == "on"
+    end
```

Comment:
@agrare I prefer this implementation to [this one](https://github.com/ManageIQ/manageiq-providers-openstack/pull/716/files#diff-2a90605fccfd8b7125cea2c62ed6e80df9d243f854c45a39f50e32ab41b48500R4-R8).  Can normalize all of them to the one you prefer to maintain.

Reply:
Downside to this implementation is that we can be evaluating needless conditions (if the first one fails) and calling `unsupported_reason_add` multiple times.  Now leaning towards the other implementation with [this one](https://github.com/ManageIQ/manageiq/blob/master/app/models/vm/operations/lifecycle.rb#L11-L16) being the easiest-to-read example for me.

Reply:
@chessbyte agree the `msg ||= ""` is easier to read and won't run multiple times like this will

I always found `unsupported_reason_add(:feature` to be awkward, for one you're already in a `supports :feature` block so having to specify the feature again seems redundant (and in your linked example, prone to bugs https://github.com/ManageIQ/manageiq/blob/master/app/models/vm/operations/lifecycle.rb#L15 retire vs publish)

I wonder if we could yield the `unsupported[feature]` to the block so that this could look like:
```
supports :standby_guest do |unsupported|
  unsupported ||= unsupported_reason(:control) unless supports?(:control)
  unsupported ||= _("The VM is not powered on") unless current_state == "on"
end
```

@Fryguy thoughts?

Reply:
Love it

Reply:
Let me try to implement that

Reply:
You could also go the `msg ||=` route here and I can look at changing how setting unsupported works separately

Reply:
@agrare I updated this PR with the `msg ||= ` approach.

Also, created https://github.com/ManageIQ/manageiq/pull/21269 for the new `supports` syntax